routing/index_graph_starter.cpp

```diff
@@ -76,10 +100,46 @@ void IndexGraphStarter::RedressRoute(vector<Joint::Id> const & route,
   }
 }
 
+void IndexGraphStarter::AddZeroLengthOnewayFeature(Joint::Id from, Joint::Id to)
+{
+  if (from == to)
+    return;
+
+  m_jointsOfAdditionalEdges.insert(from);
+  m_jointsOfAdditionalEdges.insert(to);
+  m_additionalZeroEdges.emplace_back(from, to, m_additionalNextFeatureId++);
+}
+
+void IndexGraphStarter::GetAdditionalEdgesList(Joint::Id jointId, bool isOutgoing,
+                                               vector<JointEdge> & edges)
+{
+  if (m_jointsOfAdditionalEdges.count(jointId) == 0)
+    return;
+
+  for (DirectedEdge const e : m_additionalZeroEdges)
+  {
+    if (isOutgoing)
+    {
+      if (e.GetFrom() == jointId)
+        edges.emplace_back(e.GetTo(), 0 /* weight */);
```

Comment:
Ok. I see.

|m_jointsOfAdditionalEdges| and |m_additionalZeroEdges| are filled only in AddZeroLengthOnewayFeature(Joint::Id from, Joint::Id to).

AddZeroLengthOnewayFeature() are called only from AddAdditionalZeroEdges() where AddZeroLengthOnewayFeature() connects with zero edges joints which are clones to each other.

To be on the safe side I'm adding a ASSERT:
```c++
  for (DirectedEdge const e : m_additionalZeroEdges)
  {
    ASSERT_EQUAL(GetPoint(e.GetFrom()), GetPoint(e.GetTo()), ());
...
```